src/metrics/metrics.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#ifdef HAVE_PROMETHEUS
+#include <prometheus/counter.h>
+#include <prometheus/exposer.h>
+#include <prometheus/registry.h>
+#endif
+
+namespace metrics {
+
+class MetricCounter {
+public:
+#ifdef HAVE_PROMETHEUS
+	explicit MetricCounter(prometheus::Counter *counter = nullptr) : counter_(counter) {}
+
+	inline void increment(double n = 1) const {
+		if (counter_ != nullptr) {
```

Comment:
Please consider also checking if changing this `nullptr` for a constant named can improve code readability.

Reply:
Ditto

Reply:
it looks same but is not
that is exactly the point...er.
that if we use `nullptr` without naming a reason, despite of using them for different reasons, 
then we decrease readability.
Please think why you use `nullptr` and name it, to make reading of your code smoother.